EIPs/eip-0001.md

```diff
@@ -0,0 +1,113 @@
+---
+eip: 0001
+title: ERC-1 ecrecover
+author: Viktor Pulyak <viktor.pulyak@pixelplex.io>
+type: Standards Track
+category: ERC
+status: Draft
+created: 2019-11-28
+---
+
+## Simple Summary
+
+Modification of the `ecrecover` method to restore an Echo account from
+an ECDSA signature.
+
+## Motivation
+
+Minimize the changes necessary when transferring contracts from Ethereum
+to Echo.
+
+## Description of the problem
+
+The solidity `ecrecover` method cannot be used in Echo EVM in the
+version that is assumed by default.
+
+There are three main reasons for this:
+
+##### 1. EdDSA Instead Of ECDSA
+
+Echo uses a different type of signature (EdDSA instead of ECDSA) which
+does not support recovering a public key from a signature. Accordingly,
+the mechanism cannot work by default, since `EcRecovery` returns the
+address who signed the message, and it is impossible to get it from the
+Echo signature
+
+##### 2. Special Address Format
+
+Addresses in Echo EVM is converted account ID. An account with id
+`1.2.100` will have the address
+`0x0000000000000000000000000000000000000064` which cannot be obtained
+from the signature. For this reason, even if the ECDSA signature was
+still supported, we would get problems comparing the sender address and
+the address obtained from the signature. Accordingly, comparison and
+validation (for which the `ecrecover` method is most often used) would
+not work.
+
+##### 3. Authority Mechanism
+
+Since Echo supports an advanced mechanism for setting access rights to
+an account, the private keys of the account can change
+
+## Description of the proposal
+
+The idea is to map the real address of the account and the ECDSA address
+corresponding to the account private key. When calling the contract
+method, which accepts the signature for subsequent verification, use the
+ECDSA signature. Inside the contract, restore the address from the
+signature and use the mapping to get the final address of the account.
+
+## Specification
+
+### General flow
+
+1. User generates a new pair of private and public key ECDSA and the
+   address corresponding to them
+2. User sends the received address to the blockchain, thereby linking
+   the address and an account on the Echo network
+3. User sig data with ECDSA private key and sent to the contract
+4. When using the `ecrecover` method by the contract, Echo determines
+   the address obtained from the signature, finds the address and
+   account match, and returns the address of the Echo account instead of
+   the original address.
+
+### Implementation Details
+
+#### Private Key Generation
```

Comment:
We also have account_auths in account authority. Maybe they should be also used in this mechanism?

Reply:
it's still a private key, just in the hands of another person

Reply:
So we need private keys of all specified accounts(and their "parents" if we specify account_auth) to create address?

Reply:
yes